modin/pandas/indexing.py

```diff
@@ -528,6 +528,8 @@ def __getitem__(self, key):
         """
         if callable(key):
             return self.__getitem__(key(self.df))
+        if is_tuple(key) and callable(key[0]):
+            return self.__getitem__((key[0](self.df),) + key[1:])
```

Comment:
I agree that unpacking the tuple should be done in parse_tuple, but then we should also eliminate the recursive call. I also moved the function call inside parse_tuple() so that it can be shared by LocIndexer and iLocIndexer.